core/src/env/srml/types.rs

```diff
@@ -31,14 +31,34 @@ use parity_codec::{
 
 /// The SRML fundamental types.
 #[allow(unused)]
+#[cfg_attr(feature = "test-env", derive(Debug, Clone, PartialEq, Eq))]
 pub enum DefaultSrmlTypes {}
 
+/// Empty enum for default Call type, so it cannot be constructed.
+/// For calling into the runtime, a user defined Call type required.
+/// See https://github.com/paritytech/ink-types-node-runtime.
+///
+/// # Note
+/// Some traits are only implemented to satisfy the constraints of the test environment, in order to
+/// keep the code size small.
+#[cfg_attr(feature = "test-env", derive(Debug, Clone, PartialEq, Eq))]
+pub enum Call {}
+impl parity_codec::Encode for Call {}
+
+#[cfg(feature = "test-env")]
+impl parity_codec::Decode for Call {
+    fn decode<I: parity_codec::Input>(_value: &mut I) -> Option<Self> {
+        unimplemented!()
```

Comment:
This would crash with the message `not yet implemented` but we would never intend to have an implementation for it. So I think it is better to just return `None` and add a comment describing our intent to never implement it.